<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>SynGenome | About</title>
    <link rel="preconnect" href="https://fonts.googleapis.com">
    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
    <link href="https://fonts.googleapis.com/css2?family=Roboto+Mono:ital,wght@0,100..700;1,100..700&family=Space+Mono:ital,wght@0,400;0,700;1,400;1,700&display=swap" rel="stylesheet">
    <link rel="stylesheet" href="syngenome.css">
    <link rel="icon" href="/images/logo_dna_green.svg" type="image/x-icon">
  </head>

  <body>
    <div id="navbar"></div>

    <div class="about-container">

      <div class="about-content">

<h1 id="about">About</h1>
<p>DNA is the fundamental layer of information in all living organisms. Researchers have made tremendous progress in developing technologies for reading and writing DNA, but composing new DNA sequences remains a formidable challenge. Advanced AI models, trained on genomic data from millions of organisms, now enable us to compose new genomic sequences beyond what has been generated by natural evolution.</p>
<p><strong>SynGenome is a first of its kind genomics database consisting of DNA sequences generated by Evo, a genomic language model.</strong> Evo is similar to models of human language that will generate a sequence based on a userâ€™s prompt. Given a <em>prompt</em> DNA sequence, Evo will generate a <em>response</em> that is its continuation of the genomic sequence. Essentially, Evo enables â€œautocompleteâ€ for the genome.</p>
<p>The genomes of bacteria and other prokaryotic organisms contain clusters of genes with shared functions that appear next to each other on the DNA sequence. Because of this, prompting Evo with a gene involved in a function-of-interest will often generate another gene with the same, or a related, function. We exploit this phenomenon to do <strong>function-guided generative design by â€œprompt engineeringâ€ a genomic language model</strong>.</p>
<p>SynGenome is organized according to the known functions, domains, and species of the prompt sequences. The corresponding response sequences are likely enriched for genes with those functions or domains, but they could contain many other interesting genes as well. The generated sequences in <strong>SynGenome can be very different from anything found in nature while still performing interesting biological functions</strong> (TODO: ADD LINK TO PAPER), opening a new universe of biological discovery.</p>
<h2 id="questions">Questions</h2>
<h4 id="how-was-syngenome-made-">How was SynGenome made?</h4>
<p>SynGenome contains more than 100 billion DNA base pairs generated by Evo, a DNA language model trained on millions of prokaryotic genomes. Evo is trained to predict the next token in a DNA sequence, which we can use to enable Evo to generate a response sequence that follows a DNA prompt. To prompt Evo, we used DNA sequences corresponding to known genes from prokaryotic organisms (as compiled by the <a href="https://www.uniprot.org/">UniProt database</a>). While the prompt sequences come from natural genomes, the responses can be very different from any DNA sequence found in nature.</p>
<h4 id="what-do-the-function-domain-and-species-labels-mean-">What do the â€œfunction,â€ â€œdomain,â€ and â€œspeciesâ€ labels mean?</h4>
<p>The entries in Evo are organized by the additional labels associated with the prompt sequence as provided by UniProt. Functional annotations take the form of Gene Ontology (GO) terms. Protein domain annotations are based on InterPro identifiers. And species labels correspond to the organism with a genome that contains the prompt sequence.</p>
<h4 id="how-should-i-use-syngenome-">How should I use SynGenome?</h4>
<p>Many important biological systems, with applications in genome editing or DNA synthesis, have been found by searching through genomic databases in a process referred to as â€œgenome mining.â€ The sequences in SynGenome could be mined similarly. Because the synthetic sequences are likely enriched for functions that are related to the prompt sequence, these sequences could form the basis of a medium- or high-throughput functional screening library.</p>
<h2 id="methodology">Methodology</h2>
<h4 id="prompt-construction-aditi-a-schematic-figure-would-be-good-here-">Prompt construction (@Aditi a schematic figure would be good here)</h4>
<p>Each prompt is associated with a coding sequence with a UniProt entry. We construct six prompts, each with an associated label found in the raw data:</p>
<ul>
<li><em>Upstream:</em> The X base pair (bp) sequence at the 5â€™ end of the coding sequence (CDS).  </li>
<li><em>CDS:</em> The X bp sequence beginning with the CDS.  </li>
<li><em>Downstream:</em> The X bp sequence at the 3â€™ end on the CDS.  </li>
<li><em>RevComp Upstream:</em> The reverse complement of the X base pair (bp) sequence at the 5â€™ end of the CDS.  </li>
<li><em>RevComp:</em> The X bp sequence that ends with   </li>
<li><em>RevComp Downstream:</em></li>
</ul>
<h4 id="model-and-sampling-algorithm">Model and sampling algorithm</h4>
<p>We use the Evo 1.5 model (TODO: add link) without any additional finetuning or post-training. We use a standard autoregressive decoding algorithm to sample new sequences with TODO: add sampling parameters here. Code for generating sequences with Evo can be found <a href="https://github.com/evo-design/evo/blob/main/evo/generation.py">here</a>. For each prompt as described above, we sampled X sequences.</p>
<h4 id="post-processing">Post-processing</h4>
<p>After sampling, we ran <a href="https://www.ncbi.nlm.nih.gov/IEB/ToolBox/CPP_DOC/lxr/source/src/app/dustmasker/">NCBI dustmasker</a> with default parameters (TODO: check params) to remove highly repetitive sequences, for example, long stretches of sequence containing only a single base pair. While many repetitive sequences were eliminated, we chose not to filter out all repetitive elements given that some prokaryotic genomes do naturally contain repetitive regions.</p>
<h4 id="raw-data-format">Raw data format</h4>
<p>The raw SynGenome data is provided in CSV format. Below are the column names and a brief description of the field. Sequences generated by Evo are in <strong>bold</strong>.</p>
<ul>
<li><em>UUID:</em> Unique identifier for each entry in the dataset  </li>
<li><em>Prompt:</em> Input text used to generate sequences  </li>
<li><strong><em>Generated_Seq:</em></strong> <strong>DNA sequence output generated from the prompt</strong>  </li>
<li><em>Score:</em> Evo log-likelihood of the sequence  </li>
<li><em>File_Derivation:</em> Source file  </li>
<li><em>UniProt_CID:</em> ID of the protein  </li>
<li><em>Type:</em> Classification of the prompt (e.g., Upstream, Downstream, CDS)  </li>
<li><em>Entry:</em> UniProt accession  </li>
<li><em>Organism:</em> Name of the biological organism  </li>
<li><em>Gene Ontology IDs:</em> Unique identifiers from Gene Ontology database (e.g., GO:0016020)  </li>
<li><em>Gene Ontology (biological process):</em> GO terms describing biological processes  </li>
<li><em>Gene Ontology (cellular component):</em> GO terms describing cellular locations  </li>
<li><em>Gene Ontology (GO):</em> All Gene Ontology annotations  </li>
<li><em>Gene Ontology (molecular function):</em> GO terms describing molecular activities  </li>
<li><em>Protein families:</em> Classification of protein family membership  </li>
<li><em>CDD:</em> CDD domain annotations  </li>
<li><em>DisProt:</em> DisProt database annotations  </li>
<li><em>Gene3D:</em> Gene3D database annotations  </li>
<li><em>HAMAP:</em> HAMAP database annotations  </li>
<li><em>InterPro:</em> InterPro domain annotations  </li>
<li><em>NCBIfam:</em> NCBI protein family annotations  </li>
<li><em>PANTHER:</em> PANTHER database annotations  </li>
<li><em>Pfam:</em> Pfam database annotations  </li>
<li><em>PIRSF:</em> PIRSF database classifications  </li>
<li><em>PRINTS:</em> PRINTS database annotations  </li>
<li><em>PROSITE:</em> PROSITE database annotations  </li>
<li><em>SFLD:</em> SFLD database annotations  </li>
<li><em>SMART:</em> SMART database annotations  </li>
<li><em>SUPFAM:</em> SUPFAM database annotations  </li>
<li><em>Domains Compiled:</em> Compilation of domain annotations from various sources  </li>
<li><em>Gene Names:</em> Official or common names of genes  </li>
<li><em>Protein names:</em> Names or descriptions of proteins  </li>
<li><strong><em>Generation_Proteins:</em></strong> <strong>Generated protein sequences contained in the DNA response</strong></li>
</ul>
<h2 id="limitations">Limitations</h2>
<p>When using SynGenome, please keep in mind a few important details. All of the generated DNA sequences are synthetic sequences and any downstream functional claims regarding these sequences should be based on additional experimental data. The â€œfunction,â€ â€œdomain,â€ and â€œspeciesâ€ labels in SynGenome are based on the UniProt annotations of the prompt sequence. While the response sequences may be enriched for the same GO terms or InterPro domains, there could also be a diversity of biological structures and functions contained in the response. Generating sequences with a language model is prone to highly repetitive generations; while many trivially repetitive regions have been filtered out, there are many repetitive sequences with more complex repeated motifs that have been retained.</p>
<h2 id="acknowledgements">Acknowledgements</h2>
<p>SynGenome makes use of and is inspired by other biological sequence databases such as the <a href="https://www.uniprot.org/">UniProt database</a> of protein sequences, the <a href="https://geneontology.org/">Gene Ontology database</a> of gene annotations, and the <a href="https://www.ebi.ac.uk/interpro/">InterPro database</a> of protein domain annotations. Evo 1.5 is trained on data from the <a href="https://gtdb.ecogenomic.org/">Genome Taxonomy Database (GTDB)</a> and the <a href="https://genome.jgi.doe.gov/portal/IMG_PR/IMG_PR.home.html">IMG/PR</a> and <a href="https://img.jgi.doe.gov/cgi-bin/vr/main.cgi">IMG/VR</a> databases from the <a href="https://jgi.doe.gov/">Joint Genome Institute</a>. We extend our gratitude to the developers and maintainers of these resources.</p>
<h2 id="license-and-citation">License and citation</h2>
<p>SynGenome is freely available under an <a href="https://www.apache.org/licenses/LICENSE-2.0">Apache 2.0 license</a>. If this database or any of its contents prove useful for your research, please cite Merchant et al. (2024).</p>

      </div>

      <div class="about-toc">
        <ul>
          <li><a href="#about">About</a></li>
          <li><a href="#questions">Questions</a></li>
          <li><a href="#methodology">Methodology</a></li>
          <li><a href="#limitations">Limitations</a></li>
          <li><a href="#acknowledgements">Acknowledgements</a></li>
          <li><a href="#license">License and citation</li>
        </ul>
      </div>

      <script src="assets/js/syngenome.js"></script>
      <script>
        renderNav('./navbar.html');

        document.querySelectorAll('.about-toc a').forEach(anchor => {
            anchor.addEventListener('click', function (e) {
                e.preventDefault();

                document.querySelector(this.getAttribute('href')).scrollIntoView({
                    behavior: 'smooth'
                });
            });
        });

        window.addEventListener('scroll', function() {
            const element = document.querySelector('.about-toc');
            const scrollY = window.scrollY; // The current vertical scroll position.
            const targetOffset = 40; // The final offset from the top.
            const initialOffset = 150; // The initial offset from the top.
            const scrollThreshold = initialOffset - targetOffset;

            if (scrollY < scrollThreshold) {
                // Gradually change the top position as you scroll down.
                const newTop = initialOffset - (initialOffset - targetOffset) * (scrollY / scrollThreshold);
                element.style.top = `${newTop}px`;
            } else {
                element.style.top = `${targetOffset}px`;
            }
        });
        
      </script>
  </body>
</html>
