<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>SynGenome | About</title>

    <meta property="og:url" content="https://evodesign.org/syngenome/about"/>
    <meta property="og:type" content="website"/>
    <meta property="og:title" content="SynGenome | About"/>
    <meta property="og:description" content="100 billion base pairs of AI-generated genomic sequence" />
    <meta property="og:image" content="https://evodesign.org/syngenome/images/syngenome_logo.png" />

    <meta name="twitter:card" content="summary_large_image" />
    <meta name="twitter:title" content="SynGenome | About" />
    <meta name="twitter:description" content="100 billion base pairs of AI-generated genomic sequence"/>
    <meta name="twitter:image" content="https://evodesign.org/syngenome/images/syngenome_logo.png" />
    
    <link rel="preconnect" href="https://fonts.googleapis.com">
    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
    <link href="https://fonts.googleapis.com/css2?family=Roboto+Mono:ital,wght@0,100..700;1,100..700&family=Space+Mono:ital,wght@0,400;0,700;1,400;1,700&display=swap" rel="stylesheet">
    <link rel="stylesheet" href="/syngenome/syngenome.css">
    <link rel="icon" href="/syngenome/images/logo_dna_green.svg" type="image/x-icon">

    <!-- Google tag (gtag.js) -->
    <script async src="https://www.googletagmanager.com/gtag/js?id=G-6GND0T0PXG"></script>
    <script>
      window.dataLayer = window.dataLayer || [];
      function gtag(){dataLayer.push(arguments);}
      gtag('js', new Date());

      gtag('config', 'G-6GND0T0PXG');
    </script>
  </head>

  <body>
    <div id="navbar"></div>

    <div class="about-container">

      <div class="about-content">

        <h1 id="about"><strong>SynGenome</strong><br/> Going beyond the observable evolutionary universe</h1>

    <p><strong>DNA is the fundamental layer of biological information in all living organisms.</strong> Researchers have made tremendous progress in developing technologies for reading, writing, and editing DNA, but composing new DNA sequences remains a formidable challenge. Advanced AI models, trained on genomic data from millions of organisms, now enable us to compose new genomic sequences beyond what has been generated by natural evolution.</p>

    <p><strong>SynGenome is a first-of-its-kind database consisting of synthetic DNA sequences generated by <a href="https://github.com/evo-design/evo/">Evo, a genomic language model</a>.</strong> Evo is similar to models of human language that learn to predict the next word in a sentence. Given a DNA sequence <em>prompt</em>, Evo will generate a DNA sequence <em>response</em> that continues the genomic sequence. Essentially, Evo enables &ldquo;autocomplete&rdquo; for the genome.</p>

    <p>In the genomes of prokaryotes and phage, genes with related functions frequently appear directly next to each other along the DNA sequence. As a result, prompting Evo with a sequence encoding a function of interest instructs the model to generate functionally related genes. This enables <strong>function-guided generative design by prompt engineering a genomic language model.</strong></p>

    <p>SynGenome is organized according to the known functions, domains, and species of the prompt sequences. The corresponding response sequences are likely enriched for genes with related functions or domains, but they could contain many other interesting genes as well. The <strong>generated sequences in SynGenome <a href="https://www.biorxiv.org/content/10.1101/2024.12.17.628962">may be very different from anything found in nature while still performing useful biological functions</a></strong>, opening up a new universe of biological discovery.</p>

    <h2 id="faqs">FAQs</h2>

    <h4>How was SynGenome made?</h4>

    <img src="/syngenome/images/syngenome_explainer.png" style="width: 85%;" />

    <p>SynGenome contains more than 100 billion DNA base pairs generated by Evo, a DNA language model trained on millions of prokaryotic genomes. Evo is trained to predict the next base pair in a DNA sequence, which enables Evo to generate a response sequence that follows a DNA prompt. To prompt Evo, we used DNA sequences encoding or directly adjacent to known protein-coding genes from prokaryotic organisms and bacteriophages (as compiled by the <a href="https://www.uniprot.org/">UniProt database</a>). While the prompt sequences come from natural genomes, the responses can be very different from any DNA sequence found in nature.</p>

    <h4>What do the &ldquo;function,&rdquo; &ldquo;domain,&rdquo; and &ldquo;species&rdquo; labels mean?</h4>

    <p>SynGenome entries are organized by the labels associated with the prompt sequence as provided by UniProt. Functional annotations take the form of Gene Ontology (GO) terms. Protein domain annotations are based on InterPro identifiers. Finally, the species labels correspond to the organism from which the prompt sequence was derived.</p>

    <h4>How should I use SynGenome?</h4>

    <p>Many important biological systems, with applications in genome editing or DNA synthesis, have been found by searching through genomic databases in a process referred to as &ldquo;genome mining.&rdquo; The sequences in SynGenome could be mined similarly. But rather than mining natural sequences, these sequences are generated based on their relationship with known genes in the prompt and can be unlike natural sequences, which is why <strong>we refer to this process as &ldquo;semantic mining.&rdquo;</strong> Because the synthetic sequences are likely enriched for functions that are related to the prompt sequence, these sequences could form the basis of a functional screening library.</p>

    <h4>Have any sequences in SynGenome been experimentally validated?</h4>

    <p>Using the same strategy for generating sequences in SynGenome, we have had high experimental success rates in low-throughput experiments in which we test only tens of variants for the desired function encoded by a genomic prompt. This includes the design of proteins with anti-CRISPR activity, as well as the design of both bacterial toxins and their conjugate antitoxins. See <a href="https://www.biorxiv.org/content/10.1101/2024.12.17.628962">our paper</a> for more details.</p>

    <h2 id="methodology">Methodology</h2>

    <h4>Prompt construction</h4>

    <p>Each prompt is associated with the coding sequence (CDS) of a <a href="https://www.uniprot.org/o">UniProt</a> entry. We constructed six prompts denoted in the raw data with different labels. A diagram illustrating these prompts and a table describing the prompt type labels are below.</p>

    <img src="/syngenome/images/prompt_diagram.png" />

    <table style="margin-top: 25px;">
      <tr>
        <th>Prompt type</th>
        <th>Description</th>
      </tr>
      <tr>
        <td>Upstream</td>
        <td>The 500 base pair (bp) sequence at the 5' end of the coding sequence (CDS).</td>
      </tr>
      <tr>
        <td>CDS</td>
        <td>The 500 bp sequence beginning with the CDS. If the CDS sequence is shorter than 500 bp, the CDS sequence is rounded down to the nearest 100 bp.</td>
      </tr>
      <tr>
        <td>Downstream</td>
        <td>The 500 bp sequence at the 3' end on the CDS.</td>
      </tr>
      <tr>
        <td>RevComp Upstream</td>
        <td>The reverse complement of the 500 base pair sequence at the 5' end of the CDS.</td>
      </tr>
      <tr>
        <td>RevComp</td>
        <td>The reverse complement of the 500 base pair sequence that ends with the 5' end of the CDS. If the reverse complement of the CDS sequence is shorter than 500 bp, the CDS sequence is rounded down to the nearest 100 bp.</td>
      </tr>
      <tr>
        <td>RevComp Downstream</td>
        <td>The reverse complement of the 500 base pair sequence at the 3' end of the CDS.</td>
      </tr>
    </table>

    <h4>Model and sampling algorithm</h4>

    <p>We generate sequences with the <a href="https://github.com/evo-design/evo/">Evo 1.5 model</a> without any additional finetuning or post-training. We use a standard autoregressive decoding algorithm to sample new sequences with a temperature of 0.7, top-<em>k</em> of 4, and top-<em>p</em> of 1. Code for generating sequences with Evo can be found <a href="https://github.com/evo-design/evo/blob/main/evo/generation.py">here</a>. For each prompt type described above, we sampled 2 sequences.</p>

    <h4>Post-processing</h4>

    <p>After sampling, we ran <a href="https://www.ncbi.nlm.nih.gov/IEB/ToolBox/CPP_DOC/lxr/source/src/app/dustmasker/">NCBI dustmasker</a> to remove highly repetitive sequences, for example, long stretches of sequence containing only a single base pair. While many repetitive sequences were eliminated, we chose not to filter out all repetitive elements given that some prokaryotic genomes do naturally contain biologically meaningful repeats.</p>

    <h4>Raw data format</h4>

        <p>The raw SynGenome data is provided in CSV format. Below are the column names and a brief description of the field. Fields corresponding to sequences generated by Evo are in <strong>bold</strong>.</p>

        <table>
          <tr>
            <th>Field</th>
            <th>Description</th>
          </tr>
          <tr>
            <td>UUID</td>
            <td>Unique identifier for each entry in the dataset</td>
          </tr>
          <tr>
            <td>Prompt</td>
            <td>Input text used to generate sequences</td>
          </tr>
          <tr>
            <td><strong>Generated_Seq</strong></td>
            <td><strong>DNA sequence output generated from the prompt</strong></td>
          </tr>
          <tr>
            <td>Score</td>
            <td>Evo log-likelihood of the sequence</td>
          </tr>
          <tr>
            <td>File_Derivation</td>
            <td>Source file</td>
          </tr>
          <tr>
            <td>UniProt_CID</td>
            <td>ID of the protein</td>
          </tr>
          <tr>
            <td>Type</td>
            <td>Classification of the prompt (e.g., Upstream, Downstream, CDS)</td>
          </tr>
          <tr>
            <td>Entry</td>
            <td>UniProt accession</td>
          </tr>
          <tr>
            <td>Organism</td>
            <td>Name of the biological organism</td>
          </tr>
          <tr>
            <td>Gene Ontology IDs</td>
            <td>Unique identifiers from Gene Ontology database (e.g., GO:0016020)</td>
          </tr>
          <tr>
            <td>Gene Ontology (biological process)</td>
            <td>GO terms describing biological processes</td>
          </tr>
          <tr>
            <td>Gene Ontology (cellular component)</td>
            <td>GO terms describing cellular locations</td>
          </tr>
          <tr>
            <td>Gene Ontology (GO)</td>
            <td>All Gene Ontology annotations</td>
          </tr>
          <tr>
            <td>Gene Ontology (molecular function)</td>
            <td>GO terms describing molecular activities</td>
          </tr>
          <tr>
            <td>Protein families</td>
            <td>Classification of protein family membership</td>
          </tr>
          <tr>
            <td>CDD</td>
            <td>CDD domain annotations</td>
          </tr>
          <tr>
            <td>DisProt</td>
            <td>DisProt database annotations</td>
          </tr>
          <tr>
            <td>Gene3D</td>
            <td>Gene3D database annotations</td>
          </tr>
          <tr>
            <td>HAMAP</td>
            <td>HAMAP database annotations</td>
          </tr>
          <tr>
            <td>InterPro</td>
            <td>InterPro domain annotations</td>
          </tr>
          <tr>
            <td>NCBIfam</td>
            <td>NCBI protein family annotations</td>
          </tr>
          <tr>
            <td>PANTHER</td>
            <td>PANTHER database annotations</td>
          </tr>
          <tr>
            <td>Pfam</td>
            <td>Pfam database annotations</td>
          </tr>
          <tr>
            <td>PIRSF</td>
            <td>PIRSF database classifications</td>
          </tr>
          <tr>
            <td>PRINTS</td>
            <td>PRINTS database annotations</td>
          </tr>
          <tr>
            <td>PROSITE</td>
            <td>PROSITE database annotations</td>
          </tr>
          <tr>
            <td>SFLD</td>
            <td>SFLD database annotations</td>
          </tr>
          <tr>
            <td>SMART</td>
            <td>SMART database annotations</td>
          </tr>
          <tr>
            <td>SUPFAM</td>
            <td>SUPFAM database annotations</td>
          </tr>
          <tr>
            <td>Domains Compiled</td>
            <td>Compilation of domain annotations from various sources</td>
          </tr>
          <tr>
            <td>Gene Names</td>
            <td>Official or common names of genes</td>
          </tr>
          <tr>
            <td>Protein names</td>
            <td>Names or descriptions of proteins</td>
          </tr>
          <tr>
            <td><strong>Generation_Proteins</strong></td>
            <td><strong>Generated protein sequences contained in the DNA response</strong></td>
          </tr>
        </table>

    <h2 id="limitations">Limitations</h2>

    <p>When using SynGenome, please keep in mind a few important details. All of the generated DNA sequences are synthetic sequences and any downstream functional claims regarding these sequences should be based on additional experimental data. To achieve high experimental success rates, some bioinformatic filtering related to your function of interest is recommended. The &ldquo;function,&rdquo; &ldquo;domain,&rdquo; and &ldquo;species&rdquo; labels in SynGenome are based on the UniProt annotations of the prompt sequence. While the response sequences may be enriched for the same GO terms or InterPro domains as in the prompt, there could also be a diversity of biological structures and functions contained in the response. Generating sequences with a language model is prone to highly repetitive generations; while many trivially repetitive regions have been filtered out, some repetitive sequences, especially those with more complex motifs, have been retained.</p>

    <h2 id="acknowledgements">Acknowledgements</h2>

    <p>SynGenome makes use of and is inspired by other biological sequence databases such as the <a href="https://www.uniprot.org/">UniProt database</a> of protein sequences, the <a href="https://geneontology.org/">Gene Ontology database</a> of gene annotations, and the <a href="https://www.ebi.ac.uk/interpro/">InterPro database</a> of protein domain annotations. Evo is trained on data from the <a href="https://gtdb.ecogenomic.org/">Genome Taxonomy Database (GTDB)</a> and the <a href="https://genome.jgi.doe.gov/portal/IMG_PR/IMG_PR.home.html">IMG/PR</a> and <a href="https://img.jgi.doe.gov/cgi-bin/vr/main.cgi">IMG/VR</a> databases from the <a href="https://jgi.doe.gov/">Joint Genome Institute</a>. We extend our gratitude to the developers and maintainers of these resources.</p>

    <h2 id="download">Download</h2>

    <p>The full database is available for download at Hugging Face datasets: <a href="https://huggingface.co/datasets/evo-design/syngenome-uniprot">https://huggingface.co/datasets/evo-design/syngenome-uniprot</a>.</p>

    <p>For downloads specific to a given GO term, InterPro domain, species, or UniProt ID, please use the SynGenome browse/search functionality and click the <em>&ldquo;Download prompts and generations&rdquo;</em> button on a given entry page.</p>

    <h2 id="license">License and citation</h2>

        <p>SynGenome is freely available under an <a href="https://opensource.org/license/mit">MIT license</a>. If this database or any of its contents prove useful for your research, please cite <a href="https://www.biorxiv.org/content/10.1101/2024.12.17.628962">Merchant et al. (2024)</a>.</p>

        <p class="about-citation">@article {merchant2024semantic,
    author = {Merchant, Aditi T and King, Samuel H and Nguyen, Eric and Hie, Brian L},
    title = {Semantic mining of functional de novo genes from a genomic language model},
    year = {2024},
    doi = {10.1101/2024.12.17.628962},
    publisher = {Cold Spring Harbor Laboratory},
    URL = {https://www.biorxiv.org/content/early/2024/12/18/2024.12.17.628962},
    journal = {bioRxiv}
}</p>

      <footer id="footer-links"></footer>

      </div>

      <div class="about-toc">
        <ul>
          <li><a href="#about">About</a></li>
          <li><a href="#faqs">FAQs</a></li>
          <li><a href="#methodology">Methodology</a></li>
          <li><a href="#limitations">Limitations</a></li>
          <li><a href="#acknowledgements">Acknowledgements</a></li>
          <li><a href="#download">Download</a></li>
          <li><a href="#license">License and citation</li>
        </ul>
      </div>

      <script src="/syngenome/assets/js/syngenome.js"></script>
      <script>
        renderNav('/syngenome/navbar.html');

        document.querySelectorAll('.about-toc a').forEach(anchor => {
            anchor.addEventListener('click', function (e) {
                e.preventDefault();

                document.querySelector(this.getAttribute('href')).scrollIntoView({
                    behavior: 'smooth'
                });
            });
        });

        window.addEventListener('scroll', function() {
            const element = document.querySelector('.about-toc');
            const scrollY = window.scrollY; // The current vertical scroll position.
            const targetOffset = 40; // The final offset from the top.
            const initialOffset = 150; // The initial offset from the top.
            const scrollThreshold = initialOffset - targetOffset;

            if (scrollY < scrollThreshold) {
                // Gradually change the top position as you scroll down.
                const newTop = initialOffset - (initialOffset - targetOffset) * (scrollY / scrollThreshold);
                element.style.top = `${newTop}px`;
            } else {
                element.style.top = `${targetOffset}px`;
            }
        });

        fetch('/syngenome/footer_links.html')
            .then(response => response.text())
            .then(data => document.getElementById('footer-links').innerHTML = data);
      </script>
  </body>
</html>
